yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -0,0 +1,350 @@
+		log.Info("ending the migration")
+	} else {
+		log.Info("Aborting the end migration command")
+		return
+	}
+
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		utils.ErrExit("end migration: getting migration status record: %v", err)
+	} else if msr == nil {
+		utils.ErrExit("end migration: migration status record not found. Is the migration initialized?")
+	}
+	retrieveMigrationUUID()
+	checkIfEndCommandCanBePerformed(msr)
+
+	saveMigrationReportsFn()
+
+	// cleaning only the migration state wherever and  whatever required
+	cleanupSourceDB(msr)
+	cleanupTargetDB(msr)
+	cleanupFallForwardDB(msr)
+	cleanupFallBackDB(msr)
+
+	// backing up the state from the export directory
+	backupSchemaFilesFn(msr)
+	backupDataFilesFn()
+	backupLogFilesFn()
+
+	cleanupExportDir()
+	utils.PrintAndLog("Migration ended successfully")
+}
+
+func cleanupSourceDB(msr *metadb.MigrationStatusRecord) {
+	utils.PrintAndLog("cleaning up source db...")
+	source := msr.SourceDBConf
+	if source == nil {
+		log.Infof("source db conf is not set. skipping cleanup")
+		return
+	}
+	err := source.DB().Connect()
+	if err != nil {
+		utils.ErrExit("end migration: connecting to source db: %v", err)
+	}
+	defer source.DB().Disconnect()
+	source.DB().ClearMigrationState(migrationUUID, exportDir)
+
+	err = dbzm.NewYugabyteDBCDCClient(exportDir, "", source.SSLRootCert, source.DBName, strings.Split(source.TableList, ",")[0], metaDB).DeleteStreamID()
+	if err != nil {
+		utils.ErrExit("end migration: deleting yugabytedb cdc stream id: %v", err)
+	}
+}
+
+func cleanupTargetDB(msr *metadb.MigrationStatusRecord) {
+	utils.PrintAndLog("cleaning up target db...")
+	if msr.TargetDBConf == nil {
+		log.Infof("target db conf is not set. skipping cleanup")
+		return
+	}
+	tdb := tgtdb.NewTargetDB(msr.TargetDBConf)
+	err := tdb.Init()
+	if err != nil {
+		utils.ErrExit("end migration: initializing target db: %v", err)
+	}
+	defer tdb.Finalize()
+	tdb.ClearMigrationState(migrationUUID, exportDir)
+}
+
+func cleanupFallForwardDB(msr *metadb.MigrationStatusRecord) {
+	if msr.FallForwardEnabled {
+		utils.PrintAndLog("cleaning up fall-forward db...")
+		ffdb := tgtdb.NewTargetDB(msr.FallForwardDBConf)
+		err := ffdb.Init()
+		if err != nil {
+			utils.ErrExit("end migration: initializing fallforward db: %v", err)
+		}
+		defer ffdb.Finalize()
+		ffdb.ClearMigrationState(migrationUUID, exportDir)
+	}
+}
+
+func cleanupFallBackDB(msr *metadb.MigrationStatusRecord) {
+	if msr.FallbackEnabled {
+		utils.PrintAndLog("cleaning up fallback db...")
+		fbdb := tgtdb.NewTargetDB(msr.SourceDBAsTargetConf)
+		err := fbdb.Init()
+		if err != nil {
+			utils.ErrExit("end migration: initializing fallback db: %v", err)
+		}
+		defer fbdb.Finalize()
+		fbdb.ClearMigrationState(migrationUUID, exportDir)
+	}
+}
+
+func cleanupExportDir() {
+	utils.PrintAndLog("cleaning up export dir...")
+	subdirs := []string{"schema", "data", "logs", "reports", "temp", "metainfo"}
```

Comment:
Instead of a specific list, should we delete all files under the `export-dir`?

Reply:
yeah i was doing that but the issue is it was deleting the lock file also for the `end migration` command which was causing an error while voyager looks for that file to unlock it